client/src/services/notification.ts

```diff
@@ -0,0 +1,97 @@
+import { Socket } from "socket.io-client";
+import { Notification } from '../types/notification';
+
+class NotificationService {
+    private socket: Socket;
+
+    constructor(socket: Socket) {
+        this.socket = socket;
```

Comment:
you should not be taking instance of socket inside service, should only contain inside controllers